### DISCUSSION TITLE
NodeBuilder hardcodes receiver to "this"

### DISCUSSION THREAD

**Discussion by maximiliankaul** [Fraunhofer-AISEC/cpg#510] [Q&A] [answered]
https://github.com/Fraunhofer-AISEC/cpg/blob/f065ed684301cf8fcb436fcf0faad397f5bf27b1/cpg-library/src/main/java/de/fraunhofer/aisec/cpg/graph/NodeBuilder.java#L449

The NodeBuilder's `newRecordDeclaration` always names the receiver `this`. However, python's convention (not enforced) is to name the receiver `self`. Should this be configurable?

**Comment by oxisto** [comment]
Actually, this should not really be there at all, this has bugged me for a while, that this is created as a field declaration, which does not make any sense. My workaround for Go was to NOT use the `NodeBuilder` for the class. However, we really should remove this implicit field.

**Reply by maximiliankaul** [reply]
Ok: https://github.com/Fraunhofer-AISEC/cpg/issues/511

**Comment by oxisto** [comment, marked as answer]
This is fixed by #835
